### ISSUE TITLE
bug: Error: Invalid data source when upgrading to 1.4.0

### ISSUE THREAD

**Issue by lra** [lacework/terraform-gcp-config#51]
When we use 1.4.0 we have the following error when planning:
```
╷
│ Error: Invalid data source
│ 
│   on .terraform/modules/gcp_organization_config/main.tf line 109, in data "google_folders" "my-org-folders":
│  109: data "google_folders" "my-org-folders" {
│ 
│ The provider hashicorp/google does not support data source
│ "google_folders". Did you mean "google_folder"?
╵
Error: Terraform exited with code 1.
Error: Process completed with exit code 1.
```

Going back to 1.3.0 stops the errors.

```
Terraform v1.1.7
on darwin_arm64
+ provider registry.terraform.io/hashicorp/google v3.90.1
+ provider registry.terraform.io/hashicorp/random v3.1.0
+ provider registry.terraform.io/hashicorp/time v0.7.2
+ provider registry.terraform.io/lacework/lacework v0.16.0
```

**Comment by afiune** [comment]
@lra Thank you for your feedback, we will look into this issue on our latest release.

**Comment by afiune** [comment]
@lra This data source was introduced in version 4.3 of the google provider. https://github.com/hashicorp/terraform-provider-google/releases/tag/v4.3.0

We will have to revert this change and do a major version bump since we will have to pin the version of the google provider to that version or above.

Apologies for the inconvenience.

**Comment by afiune** [comment]
We have reverted the change that broke the module and we have triggered a release that will land soon, we sincerely thank you for your prompt feedback. 🥇

**Comment by lra** [comment]
Confirmed resolved with 1.5, thanks!
